### DISCUSSION TITLE
Spectrogram period scale

### DISCUSSION THREAD

**Discussion by OndrejBocek** [audacity/audacity#4978] [Q&A] [answered]
Hey all, 

I've got a question about the spectrogram view. In the spectrogram settings you can choose various scales: linear, mel. . . and **period**. The period scale is important to me. I've been browsing the Audacity code trying to find the exact way it computes the spectrogram with the period scale and I've failed. And that's exactly what I need to know - the equation Audacity uses for the period scale. 

I'd very much appreciate any help! ^^

**Comment by Paul-Licameli** [comment, marked as answer]
One of the Algorithm choices for Spectrogram is "Pitch (EAC)" which means Enhanced AutoCorrelation, and until some old (2.1.x) versions of Audacity, it was the only choice that used a vertical scale that is essentially 1 / frequency.

In version 2.1.2 I changed things to make the choice of algorithm orthogonal to the choice of vertical scales, so the Period scale was available for other spectrogram algorithms, and the other vertical scales were available for autocorrelation.  I also defined some new vertical scales.

**Reply by OndrejBocek** [reply]
Oh yeah, that does make sense. Thanks a lot!
